File: src/lib/sRAD/gui/component/VentanaEmergente.java
package lib.sRAD.gui.component;

import lib.sRAD.gui.sComponent.SFrame;

import static lib.sRAD.gui.component.Resource.grayBorder;

public class VentanaEmergente extends SFrame {
    private final SFrame frame;

    public VentanaEmergente(SFrame frame, int width, int height) {
        this.frame = frame;
        setLayout(null);
        setProperties(width, height, Theme.veBg, true, grayBorder,null);
    }

    public void lanzar() {
        frame.setEnabled(false);
        setVisible(true);
    }

    public void cerrar() {
        frame.setEnabled(true);
        this.setVisible(false);
    }
}

File: src/lib/sRAD/kotlin/gui/component/VentanaEmergente.kt
package lib.sRAD.kotlin.gui.component

import lib.sRAD.kotlin.gui.sComponent.SFrame

open class VentanaEmergente(frame: SFrame, width: Int, height: Int): SFrame() {
    private val frame: SFrame

    init {
        this.frame = frame
        setProperties(width, height, Theme.veBg)
    }

    fun lanzar() {
        frame.isEnabled = false
        isVisible = true
    }

    fun cerrar() {
        frame.isEnabled = true
        this.isVisible = false
    }
}